internal/controller/standalone_pgadmin/pod.go

```diff
@@ -34,52 +39,81 @@ func pod(
 ) {
 	const (
 		// config and data volume names
-		configVolumeName = "standalone-pgadmin-config"
-		dataVolumeName   = "standalone-pgadmin-data"
+		mountedConfigsVolumeName = "standalone-pgadmin-config"
+		configSystemVolumeName   = "config-system"
+		dataVolumeName           = "standalone-pgadmin-data"
+		logVolume                = "pgadmin-log"
+		tempVolumeName           = "tmp"
 	)
 
-	// create the pgAdmin Pod volumes
-	pgAdminData := corev1.Volume{Name: dataVolumeName}
-	pgAdminData.VolumeSource = corev1.VolumeSource{
-		PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
-			ClaimName: pgAdminVolume.Name,
-			ReadOnly:  false,
+	// create the projected volume of config maps for use in
+	// 1. dynamic server discovery
+	// 2. adding the config variables during pgAdmin startup
+	mountedConfigsVolume := corev1.Volume{
+		Name: mountedConfigsVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			Projected: &corev1.ProjectedVolumeSource{
+				Sources: podConfigFiles(inConfigMap, *inPGAdmin),
+			},
 		},
 	}
 
-	configVolume := corev1.Volume{Name: configVolumeName}
-	configVolume.Projected = &corev1.ProjectedVolumeSource{
-		Sources: podConfigFiles(inConfigMap, *inPGAdmin),
+	// Volume used to write a custom config_system.py file in the initContainer
+	// which then loads the configs found in the `mountedConfigsVolume`
+	systemConfigsVolume := corev1.Volume{
+		Name: configSystemVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{
+				Medium: corev1.StorageMediumMemory,
+			},
```

Comment:
❓ In the other package, we seem to be limiting the size of this writable volume. I have a vague memory that it was to try to mitigate some risk that a writable executable `config_system.py` could be exploited.

https://github.com/CrunchyData/postgres-operator/blob/b05eb5433b6aa80da380926ef99935de38eb945b/internal/pgadmin/reconcile.go#L210-L218